src/HybridAppSignIn.tsx

```diff
@@ -0,0 +1,57 @@
+import {findFocusedRoute} from '@react-navigation/native';
+import {useContext, useEffect, useState} from 'react';
+import {Linking} from 'react-native';
+import type {AppProps} from './App';
+import CONST from './CONST';
+import {signInAfterTransitionFromOldDot} from './libs/actions/Session';
+import Navigation, {navigationRef} from './libs/Navigation/Navigation';
+import type {Route} from './ROUTES';
+import ROUTES from './ROUTES';
+import SCREENS from './SCREENS';
+import SplashScreenStateContext from './SplashScreenStateContext';
+
+function handleHybridUrlNavigation(url: Route) {
+    const parsedUrl = Navigation.parseHybridAppUrl(url);
+
+    Navigation.isNavigationReady().then(() => {
+        if (parsedUrl.startsWith(`/${ROUTES.SHARE_ROOT}`)) {
+            const focusRoute = findFocusedRoute(navigationRef.getRootState());
+            if (focusRoute?.name === SCREENS.SHARE.SHARE_DETAILS || focusRoute?.name === SCREENS.SHARE.SUBMIT_DETAILS) {
+                Navigation.goBack(ROUTES.SHARE_ROOT);
+                return;
+            }
+        }
+        Navigation.navigate(parsedUrl);
+    });
+}
+
+function HybridAppSignIn({url, hybridAppSettings}: AppProps) {
```

Comment:
What about different name? It also has a listener that triggers on every `url` event.

What about `HybridAppUrlHandler` or `HybridAppRouter`?

Btw maybe we should perform an early-return in `useEffect` to make sure that we don't use it on Standalone NewDot

Reply:
We won't use it on standalone ND because we keep it behind CONFIG.IS_HYBRID_APP in App.tsx